docs/proposals/remediation/cluster-status-based-remedy.md

```diff
@@ -0,0 +1,383 @@
+
+#### Component build and deployment
+
+Deploy the agent component in the form of DaemonSet in the Karmada member cluster to automatically detect cluster status changes in the cluster and report the detection results to the Karmada control plane. 
+
+When reporting detection results, it is not necessary for each instance to report, and the master selection operation is performed so that the master instance is responsible for reporting the detection results.
+
+In order to access the Karmada control plane, the Karmada control plane access credentials parameter `--karmada-kubeconfig` needs to be added to the agent component assembly for the user to specify.
+
+#### Domain name resolution function detection
+
+Instances in the component will execute commands periodically (configured through the proxy component startup parameter `--detect-period`, 5s by default), and perform domain name resolution through the `nslookup kubernetes.default` command.
+
+- If the command is executed successfully, it means that this Service domain name resolution is successful;
+- If the execution of the command fails, it means that this Service domain name resolution fails.
+
+The determination of Service's domain name resolution capability is not determined by a single command execution, the component memory will record whether the domain name resolution function is normal or not previously, assuming that the `detectionFlag` is used to indicate that the value of true means normal, and the value of false means abnormal. There are the following cases:
+
+- If `detectionFlag` is true and this Service domain name resolution is successful, the Service domain name resolution function is determined to be normal;
+- If `detectionFlag` is true and this Service domain name resolution is failed, then record the current time, execute the next command, if it lasts for a period of time (configured through the proxy component startup parameter `--detect-success-threshold`, the default is 30s) Service domain name resolution fails, then it is determined that the Service domain name resolution function is abnormal;
+- If `detectionFlag` is false and this Service domain name resolution is successful, then record the current time, execute the next command, if it lasts for a period of time (configured through the proxy component startup parameter `--detect-failure-threshold`, the default is 30s) Service domain name resolution successes, then it is determined that the Service domain name resolution function is normal; 
+- If `detectionFlag` is false and this Service domain name resolution is failed, the Service domain name resolution function is determined to be abnormal;
+
+#### Reporting of detection results
```

Comment:
Will we report the `event`?

Reply:
Is it better to generate normal or warning events?

Reply:
As I mentioned above, we can analyze and design this feature in depth.